### PR TITLE
Tests for index and show in UI

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

This bundle is a very important fundamental for Sylius itself, but at the same time, it provides so many untested functionalities... It's just a shame 😄 I propose two simple UI-based tests for the resource. To speed up implementation, I've used `ApiTestCase` to load fixtures quickly, but we should think about extracting this fixtures logic (probably in the `api-test-case` itself) and use some crawler for these tests. Let me know what you think about it  🖖

### CODE REVIEW

File: src/Bundle/test/src/Entity/ScienceBook.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+declare(strict_types=1);
+
+namespace App\Entity;
+
+use Sylius\Component\Resource\Model\ResourceInterface;
+
+class ScienceBook implements ResourceInterface
+{
+    private ?int $id;
```
Comment (loic425) [inline src/Bundle/test/src/Entity/ScienceBook.php:20]:
```suggestion
    private ?int $id = null;
```

File: src/Bundle/test/src/Entity/ScienceBook.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    private ?Author $author = null;
+
+    private ?string $title = null;
+
+    public function getId()
```
Comment (loic425) [inline src/Bundle/test/src/Entity/ScienceBook.php:26]:
```suggestion
    public function getId(): ?int
```

File: src/Bundle/test/src/Entity/ScienceBook.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    {
+        return $this->id;
+    }
+
+    public function getTitle(): ?string
+    {
+        return $this->title;
+    }
+
+    public function setTitle($title): void
```
Comment (loic425) [inline src/Bundle/test/src/Entity/ScienceBook.php:36]:
```suggestion
    public function setTitle(?string $title): void
```

File: src/Bundle/test/src/Entity/ScienceBook.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    {
+        $this->title = $title;
+    }
+
+    public function getAuthorFirstName(): ?string
+    {
+        return $this->author ? $this->author->getFirstName() : null;
```
Comment (loic425) [inline src/Bundle/test/src/Entity/ScienceBook.php:43]:
```suggestion
        return $this->author?->getFirstName();
```

File: src/Bundle/test/src/Entity/ScienceBook.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    public function getAuthorLastName(): ?string
+    {
+        return $this->author ? $this->author->getLastName() : null;
```
Comment (loic425) [inline src/Bundle/test/src/Entity/ScienceBook.php:48]:
```suggestion
        return $this->author?->getLastName();
```

**Review by loic425 — CHANGES_REQUESTED** [review]

File: src/Bundle/test/src/Entity/ScienceBook.php (hunk 4 above)
Comment (vvasiloi) [reply src/Bundle/test/src/Entity/ScienceBook.php:43]: I have been waiting for this syntax for so long...
It's annoying especially when using Doctrine and everything is nullable.

**Review by loic425 — APPROVED** [review]

**Pull request merged**